### ISSUE TITLE
Retroarch mgba save problems

### ISSUE THREAD

**Issue by renegade2k82** [mgba-emu/mgba#2144]
I have a save for the zelda a link to the past/four swords and the save acts like it is blank on retroarch with mgba core but if i use the standalone mgba emulator it reads the save fine.so i dont get why this is happing and its very weird.The save is named correctly after the rom and the correct size.i matched a test save to the one i have and it seems to match.I hope this can be fixed.

**Comment by endrift** [comment]
Please upload the save file, and also which version of the core are you using?

**Comment by renegade2k82** [comment]
I am using c7dba6b version of the core witch should be the latest one.So here are 2 saves the original 64kb one and the 8kb one that i reduced in size from the original one that worked on the stand alone mgba emulator as a .sav so i dont know if the size difference matters to retroarch but neither work.here is the link https://www.dropbox.com/s/n1p1q6fmwoommzd/zelda%20savess.zip?dl=0

**Comment by renegade2k82** [comment]
Um really need help here with this issue.i already uploaded the save files and its almost been a week....

**Comment by endrift** [comment]
Sorry about that. I thin the issue is that the RA frontend decides the save should always be a big size despite the fact that mGBA standalone sees this and decides it's the wrong save type as a result. I haven't really looked at it though.

**Comment by superuser-does** [comment]
I also have this issue. I wonder if there is a theory as to why it could be happening and indeed any possible workarounds? mGBA seems to save both .srm and .sav files. To my understanding from posts online, RetroArch only takes .srm.

**Comment by endrift** [comment]
mGBA standalone only saves .sav files. The format is the same, it's the extension that's different.

**Comment by ubergeek77** [comment]
In this case, there seems to be an issue in the way mGBA is saving games, so simply renaming the extension is not sufficient.

This seems to be happening because the `.sav` files mGBA creates are of an "unknown" type. I don't know why this is, but both the old `gbaconv` C program, and mGBA's own save conversion tool, cannot recognize the type of the `.sav` files mGBA creates. In the mGBA save conversion tool, mGBA will show "No valid formats found" for affected saves.

Despite this, these strange `.sav` files will load and save just fine in a standalone desktop instance of mGBA. But since Retroarch is more strict, these saves ***will not*** load into Retroarch until they are converted, such that they are detected as a particular save type (i.e. `FLASH 215kbit`). If you try, the mGBA core in Retroarch will just act as if there was no save file at all.

Since mGBA can't recognize its own save format, I was not able to use mGBA's built in converter, but I still needed a way to convert a save file so that it would load in Retroarch. To do this, I exported a GameShark save from mGBA, then used [this online save converter](https://savefileconverter.com/#/gba/gameshark) to convert that back to a `.sav` file. The resulting file is a "proper" `.sav` file that had a correctly detected save type in both mGBA's save converter, and the `gbaconv` C program. I was then able to reanme this to a `.srm` file, and it loaded in Retroarch just fine.

I'm not an expert on GBA saves, but there seems to be some save type property that mGBA is not saving for some reason.

---

TL;DR - To reproduce the issue, start a game in mGBA, save it, then open mGBA's save converter tool and try to load an input file. If a save is affected by this issue, it will show "No valid formats detected," and thus will not load properly in Retroarch or most other emulators. I don't know if all games are affected, or what causes this.

**Comment by endrift** [comment]
Can you please _upload_ some of these "unknown" files and tell me what game they're associated with? If you're talking about the Pokémon games with RTCs, that's due to the RTC data being appended at the end. The save converter in mGBA dev builds can strip that data.

**Comment by ubergeek77** [comment]
Sure, here is a new one I just created:

[Rockman EXE 4.5 - Real Operation (Japan).sav.zip](https://github.com/mgba-emu/mgba/files/13506444/Rockman.EXE.4.5.-.Real.Operation.Japan.sav.zip)

You are right, this is an RTC game. This was created using mGBA on Windows, the stable 0.10.2 build (`mGBA-0.10.2-win64`).
